DICOM/src/Microsoft.Health.Anonymizer.Common/EncryptFunction.cs

```diff
@@ -0,0 +1,162 @@
+                byte[] result = new byte[AesIvSize + encryptedBytes.Length];
+                Buffer.BlockCopy(iv, 0, result, 0, AesIvSize);
+                Buffer.BlockCopy(encryptedBytes, 0, result, AesIvSize, encryptedBytes.Length);
+
+                return result;
+            }
+            catch (Exception ex)
+            {
+                throw new AnonymizerException(AnonymizerErrorCode.EncryptFailed, "Failed to encrypt data.", ex);
+            }
+        }
+
+        public byte[] Decrypt(string cipherText, Encoding encoding = null)
+        {
+            EnsureArg.IsNotNull(cipherText, nameof(cipherText));
+
+            if (cipherText == string.Empty)
+            {
+                return new byte[] { };
+            }
+
+            encoding ??= Encoding.UTF8;
+            var byteData = encoding.GetBytes(cipherText);
+            return Decrypt(byteData);
+        }
+
+        public byte[] Decrypt(Stream cipherStream)
+        {
+            EnsureArg.IsNotNull(cipherStream, nameof(cipherStream));
+
+            return Decrypt(StreamToByte(cipherStream));
+        }
+
+        public byte[] Decrypt(byte[] cipherBytes)
+        {
+            EnsureArg.IsNotNull(cipherBytes, nameof(cipherBytes));
+
+            if (cipherBytes.Length == 0)
+            {
+                return cipherBytes;
+            }
+
+            // Extract IV info from base64 text
+
+            if (cipherBytes.Length < AesIvSize)
+            {
+                throw new FormatException($"The input text for decryption should not be less than {AesIvSize} bytes length!");
+            }
+
+            try
+            {
+                var iv = new byte[AesIvSize];
+                Buffer.BlockCopy(cipherBytes, 0, iv, 0, AesIvSize);
+                var encryptedBytes = new byte[cipherBytes.Length - AesIvSize];
+                Buffer.BlockCopy(cipherBytes, AesIvSize, encryptedBytes, 0, encryptedBytes.Length);
+
+                // Get decryptor
+                using Aes aes = Aes.Create();
+                aes.Key = _aesKey;
+                aes.IV = iv;
+                ICryptoTransform decryptor = aes.CreateDecryptor(aes.Key, aes.IV);
+
+                using MemoryStream msDecrypt = new MemoryStream(encryptedBytes);
+                using CryptoStream csDecrypt = new CryptoStream(msDecrypt, decryptor, CryptoStreamMode.Read);
+
+                // Read the decrypted bytes from the decrypting stream
+                using var memstream = new MemoryStream();
+                csDecrypt.CopyTo(memstream);
+                return memstream.ToArray();
+            }
+            catch (Exception ex)
+            {
+                throw new AnonymizerException(AnonymizerErrorCode.EncryptFailed, "Failed to decrypt data.", ex);
+            }
+        }
+
+        private byte[] StreamToByte(Stream inputStream)
+        {
+            EnsureArg.IsNotNull(inputStream, nameof(inputStream));
+
+            inputStream.Position = 0;
```

Comment:
🟡 SUGGESTION
I think we don't need to reset the Position here. It's the caller's duty to make sure stream has desired content/position.